SnoopCompileCore/src/snoopi.jl

```diff
@@ -86,6 +94,57 @@ function _snoopi(cmd::Expr, tmin = 0.0)
     end
 end
 
+function start_deep_timing()
+    Core.Compiler.Timings.reset_timings()
+    Core.Compiler.__set_measure_typeinf(true)
+end
+function stop_deep_timing()
+    Core.Compiler.__set_measure_typeinf(false)
+    Core.Compiler.Timings.close_current_timer()
+end
+
+function finish_snoopi_deep()
+    return Core.Compiler.Timings._timings[1]
+end
+
+function _snoopi_deep(cmd::Expr)
+    return quote
+        start_deep_timing()
+        try
+            $(esc(cmd))
+        finally
+            stop_deep_timing()
+        end
+        finish_snoopi_deep()
+    end
+end
+
+macro snoopi_deep(cmd)
```

Comment:
```suggestion
export @snoopi_deep
macro snoopi_deep(cmd)
```

Reply:
I added the export to the top! :) Thanks